libs/langchain/langchain/vectorstores/vearch.py

```diff
@@ -522,6 +522,30 @@ def delete(
         ret = all(i == 0 for i in tmp_res)
         return ret
 
+    def fill_meta_info_and_content_flag(self, record):
+        content = ""
+        meta_info = {}
+        for field in record["_source"]:
+            if field == "text":
+                content = record["_source"][field]
+                continue
+            elif field == "metadata":
+                meta_info["source"] = record["_source"][field]
+                continue
+        return content, meta_info
+
+    def fill_meta_info_and_content_no_flag(self, docs_detail):
+        content = ""
+        meta_info = {}
+        for field in docs_detail:
+            if field == "text":
+                content = docs_detail[field]
+                continue
+            elif field == "metadata":
+                meta_info["source"] = docs_detail[field]
+                continue
+        return content, meta_info
```

Comment:
These two functions are exactly the same. The only difference is their inputs

```suggestion
    def fill_meta_info_and_content(self, input_dict):
        content = ""
        meta_info = {}
        if "text" in input_dict:
            content = input_dict[field]
        if "metadata" in input_dict:
            meta_info["source"] = input_dict[field]
        return content, meta_info
```

Your function calls will be 

```
content, meta_info = self.fill_meta_info_and_content(record["_source"])
```

and

```
content, meta_info = self.fill_meta_info_and_content(docs_detail)
```

respectively.

Reply:
you can further improve this function by:

```suggestion
    def fill_meta_info_and_content(self, input_dict):
        content = input_dict.get("text", "")
        meta_info = {"source": input_dict["metadata"]} if "metadata" in input_dict else {}
        return content, meta_info
```